dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/AliasAnalysis.java

```diff
@@ -77,4 +113,76 @@ public boolean mayAlias(MemoryCoreEvent a, MemoryCoreEvent b) {
         }
     }
 
+    private void printGraph(Program program, Config configuration) {
+        if (configuration.printOption == PrintOption.NONE) {
+            return;
+        }
+
+        // Nodes are identified by thread and source location (if missing, event id).
+        // A solid line marks the existence of events that must alias.
+        // A dashed line marks the existence of events that may alias.
+        final Map<String, Set<String>> mayGraph = new HashMap<>();
+        final Map<String, Set<String>> mustGraph = new HashMap<>();
+        final List<MemoryCoreEvent> events = program.getThreadEvents(MemoryCoreEvent.class);
+        for (final MemoryCoreEvent event1 : events) {
+            final String node1 = repr(event1, configuration);
+            if (node1 == null) {
```

Comment:
We should avoid adding all `Init` events into the graph by default.  I have `repr(MemoryCoreEvent,Config)` choose when an event is visible in the graph.  I will add another option for this.

Reply:
![queue-opt-alias](https://github.com/hernanponcedeleon/Dat3M/assets/5527554/ed68cc7f-0522-49fe-8b54-3126487c2bb6)
Generated from [sv-benchmarks c/pthread/queue.c](https://gitlab.com/sosy-lab/benchmarking/sv-benchmarks/-/blob/main/c/pthread/queue.c?ref_type=heads).  This does not scale well, but it may give useful insights.